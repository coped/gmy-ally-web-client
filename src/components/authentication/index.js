import Login from "components/authentication/Login";
import LoginForm from "components/authentication/LoginForm";
import SignupForm from "components/authentication/SignupForm";

export { Login, LoginForm, SignupForm };
