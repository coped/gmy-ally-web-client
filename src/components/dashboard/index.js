import Dashboard from "components/dashboard/Dashboard";

export { Dashboard };
