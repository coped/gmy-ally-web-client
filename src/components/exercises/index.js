import Exercises from "components/exercises/Exercises";

export { Exercises };
