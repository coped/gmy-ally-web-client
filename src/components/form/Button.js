import React from "react";

export default function Button({
  children,
  classList = [],
  loading,
  onClick = undefined,
}) {
  const classes = () => {
    const list = ["button", ...classList];
    return loading ? [...list, "is-loading"].join(" ") : list.join(" ");
  };
  return (
    <div className="field">
      <div className="control">
        <button className={classes()} onClick={onClick}>
          {children}
        </button>
      </div>
    </div>
  );
}
