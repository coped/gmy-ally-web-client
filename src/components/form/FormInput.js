import React from "react";

export default function Input({ label, ...rest }) {
  return (
    <div className="field">
      <label className="label">{label}</label>
      <div className="control">
        <input className="input" {...rest} />
      </div>
    </div>
  );
}
