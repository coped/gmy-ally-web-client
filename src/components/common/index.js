import Notification from "components/common/Notification";
import Button from "components/common/Button";
import PageLoader from "components/common/PageLoader";

export { Notification, Button, PageLoader };
