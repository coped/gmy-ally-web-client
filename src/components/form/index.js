import Button from "components/form/Button";
import Input from "components/form/Input";

export { Button, Input };
