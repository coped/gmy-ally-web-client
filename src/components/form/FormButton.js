import React from "react";
import { Button } from "components/common";

export default function formButton({ children, ...rest }) {
  return (
    <div className="field">
      <div className="control">
        <Button {...rest}>{children}</Button>
      </div>
    </div>
  );
}
