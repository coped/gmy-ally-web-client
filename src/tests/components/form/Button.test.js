import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { Button } from "components/form";

Enzyme.configure({ adapter: new Adapter() });

describe("Button", () => {
  const props = { classList: ["is-info"] };
  const message = "I'm a button.";

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Button {...props}>{message}</Button>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("properly renders children", () => {
    const wrapper = shallow(<Button {...props}>{message}</Button>);
    expect(wrapper.find("button").text()).toEqual(message);
  });

  it("adds classes from classList", () => {
    const classes = ["class", "another-class"];
    const wrapper = shallow(<Button classList={classes} />);
    expect(wrapper.find("button").hasClass(classes[0])).toEqual(true);
    expect(wrapper.find("button").hasClass(classes[1])).toEqual(true);
  });

  it("handles loading animation class", () => {
    const wrapper = shallow(<Button {...props}>{message}</Button>);
    wrapper.setProps({ loading: true });
    expect(wrapper.find("button").hasClass("is-loading")).toEqual(true);
    wrapper.setProps({ loading: false });
    expect(wrapper.find("button").hasClass("is-loading")).toEqual(false);
  });

  it("has a valid snapshot", () => {
    const component = renderer.create(<Button {...props}>{message}</Button>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
