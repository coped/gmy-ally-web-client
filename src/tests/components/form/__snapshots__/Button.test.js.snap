// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button has a valid snapshot 1`] = `
<div
  className="field"
>
  <div
    className="control"
  >
    <button
      className="button is-info"
    >
      I'm a button.
    </button>
  </div>
</div>
`;
